react/app-main/components/TagsSelector/TagsSelector.tsx

```diff
@@ -57,6 +57,12 @@ export const TagsSelector: React.FunctionComponent = () => {
         }
     };
 
+    React.useEffect(() => {
+        if (selectedMainTag && selectedMainTag.relatedTags.length === 0) {
+            launchSearch();
+        }
```

Comment:
This works but I think this logic should live in `onMainTagClick` just above: putting it in a `useEffect` adds an additional re-render + makes it harder to debug